Assets/MRTK/SDK/Features/UX/Scripts/InteractiveElement/Events/EventConfigurations/SelectFarEvents.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Microsoft.MixedReality.Toolkit.UI.Interaction
+{
+    /// <summary>
+    /// The event configuration for the SelectFar InteractionState.
+    /// </summary>
+    public class SelectFarEvents : BaseInteractionEventConfiguration
+    {
+        [SerializeField]
+        [Tooltip("Whether or not to register the IMixedRealityPointerHandler for global input. If Global is true, then" +
+        " events in the SelectFar state will be fired without requiring an object to be in focus. ")]
+        private bool global = false;
```

Comment:
Oh, okay so it seems like that global property applies to _just far events_ not to all events. From the demo you showed, it was actually not clear to me -- I thought that global applied to all of the events, e.g. not just far select / pointer events, but also touch events. Thinking about it, I suppose that register / unregistering for global events does indeed apply to specific handlers such as IMixedRealityPointerHandler. 

Maybe this is mostly an issue of how the UI is presented in the editor -- the "global" checkbox is at the very top, above all the states.

Reply:
The 'Global' property for the `SelectFar` state is contained in the state configuration.  This `Global` property is only referring to the `IMixedRealityPointerHandler` which is associated with the `SelectFar` state.

The intention was to allow flexibility when registering input handlers.

Here is a demonstration of the global property behavior:

| SelectFar state properties | Runtime Behavior |
|---|---|
| ![image](https://user-images.githubusercontent.com/53493796/101219081-dedc9a80-3638-11eb-9b1a-6b53bde65a67.png) |  ![SelectFarGlobalProperty](https://user-images.githubusercontent.com/53493796/101220830-4267c780-363b-11eb-8612-028e6a273edb.gif) |